ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/picklist/activity-picklist-remote-auto-complete-options.component.ts

```diff
@@ -0,0 +1,127 @@
+                class="autoCompleteOption"
+                [value]="option"
+                (click)="onValueSelect(option)"
+            >
+                <span
+                    [innerHtml]="
+                        option.optionLabel
+                            | searchHighlight
+                                : autocompleteInput.value
+                                : ignoredSymbolsInQuery
+                    "
+                ></span>
+            </mat-option>
+        </mat-autocomplete>
+    </mat-form-field>`,
+    styles: [
+        `
+            .full-width {
+                width: 100%;
+            }
+
+            .ddp-autocomplete-field ::ng-deep .mat-form-field-infix {
+                width: auto;
+            }
+        `,
+    ],
+})
+export class ActivityPicklistRemoteAutoCompleteOptionsComponent
+    extends BaseActivityPicklistQuestion
+    implements OnInit
+{
+    @Input() studyGuid: string;
+    @Input() activityGuid: string;
+    filteredGroups: ActivityPicklistNormalizedGroup[] = [];
+
+    searchValue$ = new BehaviorSubject('');
+    picklistOptions$: Observable<ActivityPicklistOption[]>;
+
+    readonly ignoredSymbolsInQuery: string[];
+
+    constructor(
+        translate: NGXTranslateService,
+        private sortPolicy: PicklistSortingPolicy,
+        @Inject('ddp.config') public config: ConfigurationService,
+        private activityService: ActivityServiceAgent
+    ) {
+        super(translate);
+        this.ignoredSymbolsInQuery =
+            this.config.picklistAutocompleteIgnoredSymbols;
+    }
+
+    ngOnInit(): void {
+        this.sortPolicy = this.shouldBeSorted
+            ? this.sortPolicy
+            : new PicklistSortingPolicy();
+
+        this.picklistOptions$ = this.searchValue$.pipe(
+            switchMap((searchValue) => {
+                const sortedOptions = this.sortPolicy.sortPicklistOptions(
+                    this.block.picklistOptions
+                );
+
+               return this.activityService.getPickListOptions(
+                    this.block.stableId,
+                    searchValue,
+                    this.studyGuid,
+                    this.activityGuid
+                );
+            })
+        );
+    }
+
+    onInput(value): void {
+        this.searchValue$.next(value);
+    }
+
+    displayAutoComplete(option: ActivityPicklistOption | string): string {
+        return typeof option === 'string' ? option : option?.optionLabel || '';
+    }
+
+    onValueSelect(
+        value: ActivityPicklistOption,
+        detail: string | null = null
+    ): void {
+        this.block.answer = value ? [{ stableId: value.stableId, detail }] : [];
+        console.log([...this.block.answer]);
+        this.valueChanged.emit([...this.block.answer]);
+    }
+
+    private get shouldBeSorted(): boolean {
+        return !this.config.notSortedPicklistAutocompleteStableIds.includes(
+            this.block.stableId
+        );
+    }
+}
```

Comment:
This file looks copy and pasted and then you applied some changes, right?
There is a bunch of sorting that happens here. At least at this time, none of it is needed. We just have to display the list of options as it comes from the server.
Can you edit, remove anything not needed and then add some comments explaining what you are doing?